### ISSUE TITLE
SWF: Serializer problems with PollForDecisionTask

### ISSUE THREAD

**Issue by mfine** [brendanhay/amazonka#101]
Without specifying any of the optional parameters, AWS's service framework returns a serializer error. The following code:

```haskell
pollDecision :: Env -> Text -> TaskList -> IO (Either Error ())
pollDecision e domain taskList =
  runAWST e $ do
    void $ send (pollForDecisionTask domain taskList)
```

Returns the Error:

```
SerializerError "SWF" "key \"code\" not present:\n{\"__type\":\"com.amazon.coral.service#SerializationException\"}"
```

If I specify the `ReverseOrder` and `MaximumPageSize` parameters, the serializer error goes away and the call works:

```haskell
pollDecision' :: Env -> Text -> TaskList -> IO (Either Error ())
pollDecision' e domain taskList =
  runAWST e $ do
    void $ send (pfdtReverseOrder .~ (Just False) $ (pfdtMaximumPageSize .~ (Just 100) $ pollForDecisionTask domain taskList))
```

If I remove either parameter, the serializer error comes back.

I'm not having any problems with the **PollForActivityTask** without setting the optional `Identity` parameter - but maybe more of the optional parameters on **PollForDecisionTask** are causing problems?

Perhaps this issue is related to #98 and sometimes empty fields causing problems?

Thanks for the awesome library - it's great to work with! I'll try and dig into the issue some more.

**Comment by brendanhay** [comment]
Thanks for opening the issue. I'll look into it shortly, but it does appear to be related to #98 at first glance.

**Comment by brendanhay** [comment]
The fixes for #98 have been merged to `develop` via 37ead0d. I don't think those changes will actually affect this issue - but additionally the serialiser errors should now have been fixed, and you'll receive the actual SWF error.

I now believe it'll either be parameter related, or a problem with the upstream service definitions not correctly annotating fields as required. Either way I'll need the actual error to proceed.
